xbmc/cores/AudioEngine/Sinks/pipewire/AESinkPipewire.cpp

```diff
@@ -348,10 +348,24 @@ void CAESinkPipewire::EnumerateDevicesEx(AEDeviceInfoList& list, bool force)
                                   streamTypes.end());
     }
 
-    if (device.m_channels.Count() == 2 && !device.m_streamTypes.empty())
+    // If DTS-HD-MA or TrueHD are configured 8 channels are needed
+    bool hasHBRFormat = false;
+    AEDataTypeList::iterator iit, iit2;
+    iit = find(device.m_streamTypes.begin(), device.m_streamTypes.end(), CAEStreamInfo::STREAM_TYPE_TRUEHD);
```

Comment:
std::find_if would be better here so we don't need to use the temporary iterators.

Reply:
Also with find_if I have to check the return value being not end() right? I don't see how that gets easier? What do I miss?

Reply:
something like this I think

```c++
bool hasHBRFormat = std::find_if(device.m_streamTypes.cbegin(), device.m_streamTypes.cend(), [](const auto& streamType)
{
  if (streamType == CAEStreamInfo::STREAM_TYPE_TRUEHD || streamType == CAEStreamInfo::STREAM_TYPE_DTSHD)
    return true;

  return false;
}) != device.m_streamTypes.cend(); 
```